config/lib/seed.js

```diff
@@ -92,3 +65,85 @@ function reportError(err) {
   console.log(err);
   console.log();
 }
+
+function performSeedUser(user) {
+
+  var User = mongoose.model('User');
+
+  return new Promise(function(resolve, reject) {
+
+    var seedUser = {
+      username: 'user',
+      password: 'User_Password1!',
+      provider: 'local',
+      email: 'user@localhost.com',
+      firstName: 'User',
+      lastName: 'Local',
+      displayName: 'User Local',
+      roles: ['user']
+    };
+
+    var seedAdmin = {
+      username: 'admin',
+      password: 'Admin_Password1!',
+      provider: 'local',
+      email: 'admin@localhost.com',
+      firstName: 'Admin',
+      lastName: 'Local',
+      displayName: 'Admin Local',
+      roles: ['user', 'admin']
+    };
+
+    var user = null;
+
+    //If production only seed admin if it does not exist
+    if (process.env.NODE_ENV === 'production') {
+      //Add Local Admin
+
+      user = new User(seedAdmin);
+
+      User.generateRandomPassphrase()
```

Comment:
@ilanbiala Generating a password doesn't have anything to do with a User; nor do we need to have a user in context when creating one. 

The reason that we wouldn't want to introduce a user to the generate password function, is so that the function can just be concerned with what it is intended to do (Autonomy). Which is to generate a random password. There could be cases where we would just want to generate a number of random passwords, and do something with them that is not tied to any User(s). And in this case, we don't even know if the User that we defined already will even make use of the password. The password & user don't (and shouldn't) care about each other until we're ready to assign the password to the user & save it.

As for the closure. I think we'd be missing out on some very key benefits of using Promises, by not introducing a closure here. Otherwise, we have to introduce some other way of handing the user off to the `saveUser` function. Thus, it might be a reason for you wanting to pass it into the generate password method; which from what I said earlier, you can gather that I deem to be breaking the notion of SOC.